src/goto-unwind/src/goto_unwind_parse_options.cpp

```diff
@@ -0,0 +1,157 @@
+    const cmdlinet &cmdline,
+    const std::string &option_name,
+    const std::string &error_message)
+  {
+    if(cmdline.isset(option_name.c_str()))
+    {
+      if(
+        auto option_value =
+          string2optional<T>(cmdline.get_value(option_name.c_str())))
+      {
+        return option_value;
+      }
+      else
+      {
+        throw invalid_command_line_argument_exceptiont{
+          error_message +
+            "(value was: " + cmdline.get_value(option_name.c_str()) + ")",
+          "--" + option_name};
+      }
+    }
+    return nullopt;
+  }
+
+  static goto_unwind_optionst from_cmdline(const cmdlinet &cmdline)
+  {
+    goto_unwind_optionst goto_unwind_options{};
+
+    if(cmdline.args.size() == 1)
+    {
+      goto_unwind_options.source_goto_binary_filepath =
+        goto_unwind_options.destination_goto_binary_filepath = cmdline.args[0];
+    }
+    else if(cmdline.args.size() == 2)
+    {
+      goto_unwind_options.source_goto_binary_filepath = cmdline.args[0];
+      goto_unwind_options.destination_goto_binary_filepath = cmdline.args[1];
+    }
+    else
+    {
+      throw invalid_command_line_argument_exceptiont{
+        "Expecting exactly one or two arguments",
+        "<Source binary> [<Destination binary>]"};
+    }
+
+    goto_unwind_options.unwind_limit =
+      parse_int_option<std::size_t>(
+        cmdline, unwind_limit_option_name, "couldn't parse as size_t")
+        .value_or(10);
+    return goto_unwind_options;
+  }
+};
+
+int goto_unwind_parse_optionst::doit()
+{
+  const auto options = goto_unwind_optionst::from_cmdline(cmdline);
+  auto goto_model = get_goto_model(options.source_goto_binary_filepath);
+  do_unwind(goto_model, options.unwind_limit);
+  write_goto_binary(
+    options.destination_goto_binary_filepath,
+    goto_model,
+    log.get_message_handler());
+  return 0;
+}
+void goto_unwind_parse_optionst::help()
+{
+  log.status() << banner_string("goto-unwind", CBMC_VERSION) << '\n'
+               << align_center_with_border("Copyright (C) 2020") << '\n'
+               << align_center_with_border("Diffblue Ltd.") << '\n' // NOLINT(*)
+               << align_center_with_border("cbmc@diffblue.com")
+               << '\n' // NOLINT(*)
+               << '\n'
+               << goto_unwind_optionst::help;
+
+  log.status() << messaget::eom;
+}
+
+goto_modelt
+goto_unwind_parse_optionst::get_goto_model(const std::string &goto_binary_path)
+{
+  auto goto_model = [&goto_binary_path, this]() {
+    if(
+      auto goto_model =
+        read_goto_binary(goto_binary_path, log.get_message_handler()))
+    {
+      return std::move(*goto_model);
+    }
+    throw invalid_source_file_exceptiont{"Failed to read goto model from " +
+                                         goto_binary_path};
+  }();
+
+  // function pointers need to be removed for unwinding to take place
+  const bool add_safety_assertion = true;
+  const bool only_remove_const_fps = false;
+  remove_function_pointers(
```

Comment:
:pick: This is kind of a surprising place to find this. Could it be moved into a separate function which is then called from `goto_unwind_parse_optionst::doit`? That way we keep all the calls to different passes in the same place.